docs/tools/testing/testing_greentea.md

```diff
@@ -178,6 +314,191 @@ This detects your attached target and drives the test. If you need to rerun the
+
+```
+mbed test --report-json json_report.json
+```
+
+##### Plain text
+
+This creates a human-friendly text summary of the test run.
+
+```
+mbed test --report-text text_report.text
+```
+
+### Test specification JSON format
+
+The Greentea test specification format decouples the tool from your build system. It provides important data, such as test names, paths to test binaries and the platform on which the binaries should run. This file is automatically generated when running tests through Mbed CLI, but you can also provide it yourself. This way you have fine-grained control over exactly which tests are run, and through which compilers.
+
+Greentea automatically looks for files called `test_spec.json` in your working directory. You can also use the `--test-spec` argument to direct Greentea to a specific test specification file.
+
+When you use the `-t` / `--target` argument with the `--test-spec` argument, you can select which "build" should be used. In the example below, you could provide the arguments `--test-spec test_spec.json -t K64F-ARM` to only run that build's tests.
+
+#### Example of test specification file
+
+In the below example, there are two defined builds:
+
+- Build `K64F-ARM` for NXP `K64F` platform compiled with `ARMCC` compiler.
+- Build `K64F-GCC` for NXP `K64F` platform compiled with `GCC ARM` compiler.
+
+Place this file in your root folder and name it `test_spec.json`.
+
+```json
+{
+    "builds": {
+        "K64F-ARM": {
+            "platform": "K64F",
+            "toolchain": "ARM",
+            "base_path": "./BUILD/K64F/ARM",
+            "baud_rate": 9600,
+            "tests": {
+                "tests-mbedmicro-rtos-mbed-mail": {
+                    "binaries": [
+                        {
+                            "binary_type": "bootable",
+                            "path": "./BUILD/K64F/ARM/tests-mbedmicro-rtos-mbed-mail.bin"
+                        }
+                    ]
+                },
+                "tests-mbed_drivers-c_strings": {
+                    "binaries": [
+                        {
+                            "binary_type": "bootable",
+                            "path": "./BUILD/K64F/ARM/tests-mbed_drivers-c_strings.bin"
+                        }
+                    ]
+                }
+            }
+        },
+        "K64F-GCC": {
+            "platform": "K64F",
+            "toolchain": "GCC_ARM",
+            "base_path": "./BUILD/K64F/GCC_ARM",
+            "baud_rate": 9600,
+            "tests": {
+                "tests-mbedmicro-rtos-mbed-mail": {
+                    "binaries": [
+                        {
+                            "binary_type": "bootable",
+                            "path": "./BUILD/K64F/GCC_ARM/tests-mbedmicro-rtos-mbed-mail.bin"
+                        }
+                    ]
+                }
+            }
+        }
+    }
+}
+```
+
+If you now run `mbed test -l` this will now list only these tests:
```

Comment:
You want to use the `--run-list` option here as well.

```suggestion
If you now run `mbed test --run-list` this will now list only these tests:
```